bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -74,7 +74,7 @@ public void initialize() {
         scheduler.execute(() -> {
             insteonBinding = new InsteonBinding(this, config, serialPortManager);
 
-            updateStatus(ThingStatus.ONLINE);
+            updateStatus(ThingStatus.UNKNOWN);
```

Comment:
I would move this to before the scheduler executes. It is generally expected that a call to `initialize()` would update the ThingStatus, even if just to UNKNOWN.

Reply:
I made the change per Kai's suggestion above:

kaikreuzer 2 days ago Member
Suggested change 
            updateStatus(ThingStatus.ONLINE);
            updateStatus(ThingStatus.UNKOWN);
You didn't have any successful communication with the bridge here, did you?

Reply:
No I have been keeping up with the conversations, but I have been keeping my input to a minimum since you seem to reject most of my suggestions. I'm sure Kai will weigh in anyway.

Reply:
I've made a fair number of your suggestions, and for this one I did what Kai suggested as well.

Reply:
@robnielsen My request was to change ONLINE to UNKNOWN, I wasn't saying that it has to be done in exactly this line.
I agree with @cpmeister that the better place is before the scheduler code, so if you'd move it, it'd be nice. It won't really harm to leave it here either, though.
In general, doing it early in the code (and in the same thread) is safer, because you are expected to set the Thing status and the later you do it, the higher the risk is that you run some code that for some reason throws an exception or does any other unexpected flow.

Reply:
@cpmeister, @kaikreuzer I moved it. Just a suggestion maybe this would be good to add to some sort of implementation guide?

Reply:
It is documented in detail directly in the code that is provided as a skeleton, including an example: https://github.com/openhab/openhab-core/blob/master/tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__Handler.java#L70-L91